src/pip/_internal/vcs/__init__.py

```diff
@@ -186,6 +187,28 @@ class VersionControl(object):
     unset_environ = ()  # type: Tuple[str, ...]
     default_arg_rev = None  # type: Optional[str]
 
+    @classmethod
+    def should_add_vcs_url_prefix(cls, remote_url):
```

Comment:
I would have named it `startswith_vcs_prefix` and used it via:
```
        if not cls.startswith_vcs_prefix(repo_url):
            repo_url = '{}+{}'.format(cls.name, repo_url)
```

Reply:
This is another case where the subclass's implementation is coming into play and affecting things. The `Subversion` class's implementation always returns `True`, so `startswith_vcs_prefix()` wouldn't accurately describe that implementation. It's possible we could change the `Subversion` class to match the other classes (and then use the name you suggest), but I wanted to avoid any behavior changes in this PR and stick to a pure refactor. I think it's best if we did that in a separate PR to keep from changing too many things at once.

Reply:
It's fine to do it in an other PR.